octodns/provider/yaml.py

```diff
@@ -207,7 +208,23 @@ def populate(self, zone, target=False, lenient=False):
             return False
 
         before = len(zone.records)
-        filename = join(self.directory, f'{zone.name}yaml')
+        utf8_filename = join(self.directory, f'{zone.decoded_name}yaml')
+        idna_filename = join(self.directory, f'{zone.name}yaml')
+
+        # we prefer utf8
+        if isfile(utf8_filename):
+            if utf8_filename != idna_filename and isfile(idna_filename):
+                raise ProviderException(
+                    f'Both UTF-8 "{utf8_filename}" and IDNA "{idna_filename}" exist for {zone.decoded_name}'
+                )
+            filename = utf8_filename
+        else:
+            self.log.warning(
+                'populate: "%s" does not exist, falling back to try idna version "%s"',
+                utf8_filename,
+                idna_filename,
+            )
+            filename = idna_filename
```

Comment:
Meaningful change. `YamlProvider` now supports reading config from either utf-8 (preferred) or idna encoded (fallback) filenames. This is to support anyone who has an existing idna encoded zone configured. It's the only known bit required for backwards compatibility since before this PR people couldn't have been doing non-ascii zones without idna encoding them manually.